pallets/gated-marketplace/src/functions.rs

```diff
@@ -830,6 +843,84 @@ impl<T: Config> Pallet<T> {
 		Ok(())
 	}
 
+	pub fn do_block_user(
```

Comment:
I like the toggle functionality but I'll recommend refactoring the whole function. Let's start using an enum, something like:
```rust
pub enum RedeemArgs<T: Config> {
	AskForRedemption { collection_id: T::CollectionId, item_id: T::ItemId, },
	AcceptRedemption (RedemptionId),
}
```
See gated-marketplace pallet. The above functionality will allow users to select if they want to block/unblock a user in a single extrinsic. 
Then, do a match like:
```rust
			match redeem {
				RedeemArgs::AskForRedemption { collection_id, item_id } => {
					return Self::do_ask_for_redeem(who, marketplace, collection_id, item_id);
				}
				RedeemArgs::AcceptRedemption ( redemption_id ) => {
					return Self::do_accept_redeem(who, marketplace, redemption_id);
				}
			}
```
Then create two separate functions: `do_block_user` & `do_unblock_user`; with the previous modularization, it is easier to read the code.  
You could choose to emit a general event in the primary function `do_block_user` but not in the helper functions or emit an event in each of the following functions: `do_block_user` & `do_unblock_user` but not in the primary function. 

```rust
	UserBlocked(MarketplaceId, T::AccountId),
	/// User was unblocked. [marketplace_id, account]
	UserUnblocked(MarketplaceId, T::AccountId),
```

Reply:
I like the idea of using an enum to chose whether to block or unblock, and i agree that having two separated helper functions helps by making the code more readable. i will use this approach.